pkg/certificate/interface.go

```diff
@@ -0,0 +1,101 @@
+package certificate
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type ProviderType string
+
+const (
+	Auto        ProviderType = "auto"
+	CertManager ProviderType = "cert-manager"
+	// add more ...
+)
+
+func NewProvider(pt ProviderType) Provider {
+	switch pt {
+	case Auto:
+		return nil // change me later
+	case CertManager:
+		return nil // change me later
+	}
+	panic(fmt.Sprintf("unknown provider type: %s", pt))
+}
+
+type Config struct {
+	CommonName       string
+	Organization     []string
+	DNSNames         []string
```

Comment:
should be extraSANs or similar as a SAN can also be an IP

Reply:
Currently we only use DNSName, as POD IP is uncertain and changeable.

We can extend it in the future when needed.

Reply:
Updated anyway now to keep it as more generic as possible. Added `AltNames` similar to what k8s.io/client-go does

https://github.com/kubernetes/client-go/blob/f2030849e1d6d961fa150b9be9560da33e5d00df/util/cert/cert.go#L51-L57